### ISSUE TITLE
Cart products are not persisting when user signout and signin again

### ISSUE THREAD

**Issue by faisal-project** [mageplaza/magento-2-social-login#201]
### Preconditions

1. Magento: 2.2.11
2. PHP: 7.2.31
3. MySQL: 10.1.46-MariaDB
4. 

### Steps to reproduce

1. Sign in through default magento login or Social login (via facebook or google)
2. Add products in the cart
3. Sign out
4. Remove the browser cache or Open the site in a private window.
5. Now Sign in through social login (facebook or google)
6. Check the cart products (it's not updating anymore)

### Expected result

1. After remove the browser cache or in a private window, when we sign in (through Social Login) again the cart products should be updated.

### Actual result

1. Now cart products is not updating, it's updating when we adding a new product in the cart.

**Comment by Victor-Mageplaza** [comment]
Hi @faisal-project 

I'm Victor from Mageplaza. I would like to answer your question.
You can refer to #197 , that topic may be related to your problem.
Then try again and let me know. Thank you!

Best Regard,
Victor

**Comment by faisal-project** [comment]
Hello @Victor-Mageplaza,

First of all thanks for your response!

I tried this #197 solution but didn't work in my case.
According to your suggestion, I pasted this code: **customerData.reload(true);** inside _create: function () { } in popup.js file, but it didn't work.

Actually this problem happening, when we login through your social login module while it's working fine with magento default login functionality.
Hope, you are understanding my point, looking for the solution, please look into this.


Best Regard,
Faisal

**Comment by faisal-project** [comment]
Hello @Victor-Mageplaza 

I have solved the problem by adding below code in **provider.js** (file path: /Mageplaza/SocialLogin/view/frontend/web/js/provider.js) file.

> var sections = ['cart'];
> customerData.invalidate(sections);
> customerData.reload(sections, true);

If anyone else facing the same issue, so they can check here: [https://prntscr.com/uky1az](url), how I have added the code.
<br>
Best Regard,
Faisal

**Comment by Victor-Mageplaza** [comment]
Hi @faisal-project 

Sorry for the late reply. First of all, I would like to thank you for taking the time to solve the module's problem.
Returning to the problem, earlier the way I suggested to you was to add **CustomerData.reload (true);** also indirectly reset the cart section.

Why is it not working? Seem strange.
I think I will test it again.
Anyway, glad to hear your issue has been resolved.
Have a nice day. Thank you!

Best Regard,
Victor
